### ISSUE TITLE
"Set body tip" seems broken

### ISSUE THREAD

**Issue by wohltat** [realthunder/FreeCAD#772]
Either the "set body tip" is broken or it is completely unintuitive. 

![set_body_tip_bug](https://user-images.githubusercontent.com/4751351/227773996-1407b50b-2d59-4f66-8a69-364a3fc6df62.gif)

I expect the sphere to be added to the cube and not to the cube+cone, after changing the body tip to the cube.
This is also what the tool tip suggests.

> Alt + click this icon to set this feature as the tip of the body.
> New feature will be inserted after it.

![image](https://user-images.githubusercontent.com/4751351/227774714-39679d09-29cc-491c-b4cb-841912b15574.png)

Furthermore the GUI shows inconsistent behavior. First the green label is only at the feature with that is the body tip. But sometimes every feature has those labels.
![image](https://user-images.githubusercontent.com/4751351/227772569-72c94f9d-e284-499f-93d7-a86f1497ebb4.png)

This is especially confusing since it appears seemly random at some point and it replaces the Attachment icon that also uses the Alt key.
![image](https://user-images.githubusercontent.com/4751351/227773384-c8a3be94-265a-408f-8c81-dd1693b3e363.png)

Btw, I know that i can move the features around by dragging and with "move object after other object". The language is also inconsistent here. Why is it called object and not feature? Calling a hole or a filled in PartDesign an 'object' doesn't make much sense. 

----------------

OS: Linux Mint 20.1 (XFCE/xfce)
Word size of FreeCAD: 64-bit
Version: 2023.131.26244 +5365 (Git) AppImage
Build type: Release
Branch: LinkDaily
Hash: 73c4ca254c5de8849d104973482e98399fb24863
Python version: 3.10.8
Qt version: 5.15.4
Coin version: 4.0.1
OCC version: 7.6.3
Locale: English/United States (en_US)

**Comment by realthunder** [comment]
Yes, body tip seems to be ineffective when adding primitive shapes. But it works for other features.

> This is especially confusing since it appears seemly random at some point and it replaces the Attachment icon that also uses the Alt key.

The tool tip is specific to the icon. `Alt + Click` the icon activates the specific action associated with the icon. I am not sure which part you find confusing, because the tag and the attachment are two very different icons.


> Furthermore the GUI shows inconsistent behavior. First the green label is only at the feature with that is the body tip. But sometimes every feature has those labels.

The tip is shown as a tag icon with center green, while all other tag icon are center white. The multiple color tag icons only show up if you have explicitly created a new shape/solid in the body, otherwise only the tip feature has the tag icon. Do you mean you've seen more than one tag icon with center green? 



> Btw, I know that i can move the features around by dragging and with "move object after other object". The language is also inconsistent here. Why is it called object and not feature? Calling a hole or a filled in PartDesign an 'object' doesn't make much sense.

Yes, I agree. But object and feature are really the same thing in FreeCAD, in a much deeper sense. For example, the `Hole` feature actually carries all shape geometry from all of its ancestor features. It is very hard to infer which of the geometry belongs to which ancestor feature.

**Comment by wohltat** [comment]
You are right, it indeed works with other features.


> The multiple color tag icons only show up if you have explicitly created a new shape/solid in the body, otherwise only the tip feature has the tag icon. Do you mean you've seen more than one tag icon with center green?

> I am not sure which part you find confusing, because the tag and the attachment are two very different icons.

Ok i see now that it appears if i create a new solid. That was mainly what confused me. That give the impression to a beginner that the label itself is the indicator for having the body tip. Instead it is the little green dot.
But it's still kind of confusing / inconsistent that why it doesn't go away if i turn the new-solid off again. Why not just have them on all the time?
Furthermore the green dot is not very easily visible. Would improve visibility if the off state is black (or better icon-color) instead of white.


> But object and feature are really the same thing in FreeCAD, in a much deeper sense. For example, the `Hole` feature actually carries all shape geometry from all of its ancestor features.

From a programmer perspective pretty much everything is an object i guess. From a user perspective the word 'object' refers more to a physical / 3D object. Furthermore it is also not consistent within itself.
If i press "move object after other object" it asks for a feature in the next window:
![image](https://user-images.githubusercontent.com/4751351/228942411-e82ae92c-7d9f-422e-992c-d909ebb8646a.png)

![image](https://user-images.githubusercontent.com/4751351/228941889-ade69f63-d51d-4c48-bf33-bce624039323.png)

**Comment by BloodyRain2k** [comment]
I agree on the confusing wording for "moving" a feature, because we're actually just reordering them.

There *is* somewhere the function to actually move something from one part to another part, but I haven't seen it in a long time. Small possiblity that it's a Main branch feature, which I haven't used in over a year, so that would explain that.
Anyways, the wording for this function is almost the exact same, but it's doing something completely different.

So I would second a rewording for the feature reordering entry. "Move object after object" sounds confusing to begin with, even when you know the context and what it does...
Maybe "Reorder feature" ?


And regarding the "Set tip" function do I also have something to add: it is not showing up for "Base" features.
![image](https://user-images.githubusercontent.com/1014362/230764598-6584ecad-e1d2-4ad9-9e21-0b52496a2c39.png)
The Part's tip can be manually set to it through the property panel, but I find it weird that the entry is missing here.
